solution/ui/regulations/eregs-vite/src/views/PolicyRepository.vue

```diff
@@ -205,58 +234,24 @@ const getDocSubjects = async () => {
 
 watch(
     () => $route.query,
-    async (newQueryParams, oldQueryParams) => {
-        // if all params are removed, clear selectedParams and getDocList
+    async (newQueryParams) => {
+        // wipe everything clean to start
+        clearSelectedParams();
+        clearSearchQuery();
+
+        // if all params are removed, getDocList with no arguments and return
         if (_isEmpty(newQueryParams)) {
-            clearSelectedParams();
             getDocList();
             return;
         }
 
-        // if there are multiple subjects already selected and one needs to be removed
-        if (
-            newQueryParams.subjects &&
-            oldQueryParams.subjects &&
-            newQueryParams.subjects.length < oldQueryParams.subjects.length
-        ) {
-            const oldSubjectIds = oldQueryParams.subjects
-                .split(",")
-                .filter((id) => !Number.isNaN(parseInt(id, 10)));
-            const newSubjectIds = newQueryParams.subjects
-                .split(",")
-                .filter((id) => !Number.isNaN(parseInt(id, 10)));
-
-            const subjectToRemove = _difference(oldSubjectIds, newSubjectIds);
-
-            removeSelectedParams({
-                type: "subjects",
-                id: subjectToRemove[0],
-            });
-
-            const newRequestParams = getRequestParams(newQueryParams);
-            await getDocList(newRequestParams);
-
-            return;
-        }
-
-        if (newQueryParams.subjects) {
-            const subjectIds = newQueryParams.subjects
-                .split(",")
-                .filter((id) => !Number.isNaN(parseInt(id, 10)));
-            const subjects = policyDocSubjects.value.results.filter((subject) =>
-                subjectIds.includes(subject.id.toString())
-            );
-
-            subjects.forEach((subject) => {
-                addSelectedParams({
-                    type: "subjects",
-                    id: subject.id,
-                    name: getSubjectName(subject),
-                });
-            });
-        }
+        // now that everything is cleaned, iterate over new query params
+        Object.entries(newQueryParams).forEach(
+            setSelectedParams(policyDocSubjects)
+        );
```

Comment:
The logic pertaining to updating the component state when the `$route` changes has been heavily refactored here.  

Before, I was trying to compare old and new query parameters and adjust the component state accordingly.  

Now, I'm just clearing component state and applying the new query parameters to that cleared state.  It's much cleaner and much less complex.